whipper/common/path.py

```diff
@@ -26,42 +26,18 @@ class PathFilter(object):
     I filter path components for safe storage on file systems.
     """
 
-    def __init__(self, slashes=True, quotes=True, fat=True, special=False):
+    def __init__(self, posix=True, fat=False):
         """
-        @param slashes: whether to convert slashes to dashes
-        @param quotes:  whether to normalize quotes
+        @param posix:   whether to strip characters illegal on POSIX platforms
         @param fat:     whether to strip characters illegal on FAT filesystems
-        @param special: whether to strip special characters
         """
-        self._slashes = slashes
-        self._quotes = quotes
+        self._posix = posix
         self._fat = fat
-        self._special = special
 
     def filter(self, path):
-        if self._slashes:
-            path = re.sub(r'[/\\]', '-', path, re.UNICODE)
-
-        def separators(path):
-            # replace separators with a space-hyphen or hyphen
-            path = re.sub(r'[:]', ' -', path, re.UNICODE)
-            path = re.sub(r'[\|]', '-', path, re.UNICODE)
-            return path
-
-        # change all fancy single/double quotes to normal quotes
-        if self._quotes:
-            path = re.sub(ur'[\xc2\xb4\u2018\u2019\u201b]', "'", path,
-                          re.UNICODE)
-            path = re.sub(ur'[\u201c\u201d\u201f]', '"', path, re.UNICODE)
-
-        if self._special:
-            path = separators(path)
-            path = re.sub(r'[\*\?&!\'\"\$\(\)`{}\[\]<>]',
-                          '_', path, re.UNICODE)
-
+        if self._posix:
+            path = re.sub(r'[\/\x00]', '_', path, re.UNICODE)
```

Comment:
I know this raises complexity, but should we also check for (and replace) `.` if it's the first character? For most (all?) POSIX systems `.*` are hidden files, and we probably don't want to "hide" ripped files.

There's [only one artist](https://musicbrainz.org/search?query=artist%3A%22.%22&type=artist&limit=25&method=advanced) but [a lot of recordings/tracks](https://musicbrainz.org/search?query=recording%3A%22.%22&type=recording&limit=25&method=advanced) (in MusicBrainz) with a "." as their first character, so while it's a small chance for someone to encounter this, it is a real case that could be encountered, even using the default settings (`DEFAULT_DISC_TEMPLATE = u'%r/%A - %d/%A - %d'` would result in `ep/. - IHI/. - IHI` for [.’s "IHI"](https://musicbrainz.org/release/fba611e2-2e17-4cae-83c7-3751f6cf734)). Granted, that person has only released digital stuff so far, but someone might want to put the release name first or do something else.

Reply:
Oh, https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations for 'POSIX "Fully portable filenames"' also says:
> hyphen must not be first character

So if we need to check for hyphen as first character, we might as well also check for period?

Reply:
> I know this raises complexity, but should we also check for (and replace) `.` if it's the first character? For most (all?) POSIX systems `.*` are hidden files, and we probably don't want to "hide" ripped files.

Done.

> hyphen must not be first character

I think this one is no longer relevant.